### PR TITLE
SALTO-2458/add field support for indexs

### PR DESCRIPTION
**Pull request by guyhamo**
after finding a way to retrieve the nacl file paths of field elements (in saas) I changed the code to add the field's element id and not his top level parent.

---

_Additional context for reviewer_

---
_Release Notes_: 

---
_User Notifications_:

### CODE REVIEW

File: packages/workspace/src/workspace/changed_at_index.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ const { isDefined } = values
 const { awu } = collections.asynciterable
 
 const log = logger(module)
```
Comment (alonstern) [inline packages/workspace/src/workspace/changed_at_index.ts:41]: unrelated question: I see we don't have `unknown` in this index. Don't we need it?

Comment (guyhamo) [reply packages/workspace/src/workspace/changed_at_index.ts:41]: nope, at least for the design there is no plan to filter by elements without a timestamp.

File: packages/workspace/src/workspace/changed_by_index.ts [hunk 2]
```diff
@@ -79,36 +82,40 @@ const getAllElementsChanges = async (
...
   }
-  if (element.annotations[CORE_ANNOTATIONS.CHANGED_BY]) {
-    authors.push(`${element.elemID.adapter}${CHANGED_BY_KEY_DELIMITER}${
-      element.annotations[CORE_ANNOTATIONS.CHANGED_BY]
-    }`)
+  if (changeElement.annotations[CORE_ANNOTATIONS.CHANGED_BY]) {
+    addElementToMap(changeElement)
   }
-  return authors.length > 0 ? authors : [UNKNOWN_USER_NAME]
+  return Object.keys(authors).length > 0 ? authors : { [UNKNOWN_USER_NAME]: [changeElement.elemID] }
```
[the remark this answers is not in the document]
Comment (guyhamo) [reply packages/workspace/src/workspace/changed_by_index.ts]: yeah before i didnt have a way to get a nacl file path from field ids so i used the object name.
now i have a way so i changed it.
I made another change in this function after noticing a bug that can happen if there was information on the field and not in the object.

**Review by alonstern — APPROVED** [review]
LGTM

**Pull request merged**